src/components/PaymentCard/PaymentCard.js

```diff
@@ -3,88 +3,126 @@ import PropTypes from 'prop-types';
 import { useTranslation } from 'react-i18next';
 import SkeletonWrapper from 'components/SkeletonWrapper';
 import { POPUP_TYPES } from 'redux/innerPopupReducer';
-import { CardTypesIcons } from './Payment.const';
+import Card from 'components/Card';
+import { CardTypes } from './Payment.const';
 import {
   CardStyled,
-  CardWrapStyled,
   CardTypeStyled,
   CardNumberStyled,
   CardExpirationStyled,
   CardExpirationLabel,
   CardExpirationDateStyled,
   CardEditStyled,
-  MethodNameStyled
+  CardHeaderStyled,
+  CardInfoStyled,
+  CardDetailsStyled,
+  CardDetailsNameStyled,
+  CardDetailsNameWrapStyled,
+  CardInfoWrapStyled
 } from './PaymentCardStyled';
 
-const INAPPS = ['android', 'apple', 'amazon', 'roku'];
+const PaymentCardSkeleton = () => (
+  <CardStyled>
+    <CardHeaderStyled>
+      <SkeletonWrapper width={140} />
+    </CardHeaderStyled>
+    <CardInfoWrapStyled>
+      <CardInfoStyled>
+        <CardTypeStyled>
+          <SkeletonWrapper height={30} width={40} />
+        </CardTypeStyled>
+        <CardDetailsStyled>
+          <CardDetailsNameWrapStyled>
+            <CardDetailsNameStyled>
+              <SkeletonWrapper width={140} />
+            </CardDetailsNameStyled>
+          </CardDetailsNameWrapStyled>
+          <CardExpirationStyled>
+            <SkeletonWrapper width={100} />
+          </CardExpirationStyled>
+        </CardDetailsStyled>
+      </CardInfoStyled>
+      <SkeletonWrapper height={43} width={170} />
+    </CardInfoWrapStyled>
+  </CardStyled>
+);
+
 const PaymentCard = ({ isDataLoaded, details, showInnerPopup }) => {
   const { t } = useTranslation();
   const { paymentMethodSpecificParams, paymentMethod } = details;
-  const LogoComponent =
-    paymentMethodSpecificParams &&
-    CardTypesIcons[paymentMethodSpecificParams.variant]
-      ? CardTypesIcons[paymentMethodSpecificParams.variant]
-      : CardTypesIcons[details.paymentMethod] || null;
+
+  let LogoComponent = null;
+  let methodTitle = '';
+  if (
+    paymentMethod === 'card' &&
+    CardTypes[paymentMethodSpecificParams.variant]
+  ) {
+    LogoComponent = CardTypes[paymentMethodSpecificParams.variant]?.icon;
+    methodTitle = CardTypes[paymentMethodSpecificParams.variant]?.title;
+  } else if (paymentMethod === 'paypal') {
+    LogoComponent = CardTypes[paymentMethod]?.icon;
+    methodTitle = CardTypes[paymentMethod]?.title;
+  } else {
+    LogoComponent = CardTypes[details.paymentMethod]?.icon;
+    methodTitle = CardTypes[details.paymentMethod]?.title;
+  }
```

Comment:
I would change it to something like
```
const specificParams = paymentMethodSpecificParams.variant || paymentMethod || details.paymentMethod;
const {icon: LogoComponent,  title: methodTitle} = CardTypes[specificParams] || {icon: null/<></> //defaultIcon, title: ''};
```
To test if this works

Reply:
That won't work because even if we have `paymentMethod === 'applepay'` there is still `paymentMethodSpecificParams.variant` set with the value `'mc_applepay'`which is not represented in the CardTypes.
I think that we can leave as it is. I know its more lines of code but still readable.
What do you think about that?

Reply:
ok, so maybe: 
```
const getSpecificParam = () => {
  if(paymentMethod === 'card') return paymentMethodSpecificParams.variant;
  if(paymentMethod === 'paypal') return paymentMethod;
  return details.paymentMethod;
}
const {icon: LogoComponent,  title: methodTitle} = CardTypes[getSpecificParam()] || {icon: null/<></> //defaultIcon, title: ''};
```
For me the initial implementation is not so readable, that's why I'm looking for different solution :)